### PR TITLE
Remove `ExpressionScriptCompilationException` and `ExpressionScriptExecutionException`

### PR DESCRIPTION
**Pull request by s1monw**
These exceptions are useless and unused, since we are on a major verison we should remove
them. This commit also makes it easier to remove exceptions in the future.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/ElasticsearchException.java [hunk 1]
```diff
@@ -463,153 +463,150 @@ public String toString() {
...
+        exceptions.put(org.elasticsearch.search.builder.SearchSourceBuilderException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.index.engine.EngineClosedException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.action.NoShardAvailableActionException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.action.UnavailableShardsException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.index.engine.FlushFailedEngineException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.common.breaker.CircuitBreakingException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.transport.NodeNotConnectedException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.index.mapper.StrictDynamicMappingException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.action.support.replication.TransportReplicationAction.RetryOnReplicaException.class, exceptions.size());
+        exceptions.put(org.elasticsearch.indices.TypeMissingException.class, exceptions.size());
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/ElasticsearchException.java]: Nice!

Why use the fully qualified class names here, btw?

**Comment by nik9000** [comment]
Left a question. LGTM though.

File: core/src/main/java/org/elasticsearch/ElasticsearchException.java (hunk 1 above)
Comment (rmuir) [inline core/src/main/java/org/elasticsearch/ElasticsearchException.java]: The fqdn was used before though too. It is not introduced here. It makes it clear which exception it is (e.g. org.elasticsearch.script.ScriptException vs javax.script.ScriptException), and it makes it easier to remove exceptions from here in master (vs having an import statement too), which is really needed.

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/ElasticsearchException.java]: Makes sense to me.

**Pull request merged**